### PR TITLE
Add storage conformance suite

### PR DESCRIPTION
**Pull request by dukesteen**
## Summary

- add a public, framework-neutral storage conformance suite selected by StorageCapabilities
- resolve IJobStorage from an IServiceProvider and verify its advertised capabilities before each scenario
- exercise in-memory, Redis, single-server, Entity Framework Core, and LinqToDB registrations through their public DI APIs
- replace redundant provider tests with 48 reusable conformance cases while retaining provider-specific race, corruption, schema, topology, and execution-strategy coverage
- consolidate provider, conformance, and storage implementation tests in Immediate.Jobs.StorageTests
- keep portable in-memory conformance and suite infrastructure coverage on .NET 8, 9, 10, and 11 while limiting the container-backed matrix to .NET 10
- keep scheduler and generated-job runtime behavior in Immediate.Jobs.FunctionalTests
- document the storage contract and provider-author workflow

This PR is stacked on #109 and targets refactor/internalize-storage-providers.

## Testing

- dotnet build Immediate.Jobs.slnx --no-restore -v minimal
- 105 functional tests passed on .NET 10
- 561 storage tests passed across SQLite, PostgreSQL, SQL Server, and Redis on .NET 10
- 248 portable storage conformance and infrastructure tests passed across .NET 8, 9, 10, and 11

## InternalsVisibleTo follow-up

The conformance suite itself no longer requires InternalsVisibleTo: it uses public registration APIs, receives an IServiceProvider, and resolves IJobStorage from the built container.

Some existing friendships remain because the surviving provider-specific white-box tests directly instantiate internal RedisJobStorage, LinqToDBJobStorage, EntityFrameworkCoreJobStorage, InMemoryJobStorage, and SingleServerJobStorage implementations. Those tests cover cross-instance races, recovery, corruption, transaction behavior, and single-server internals that are deliberately outside the portable conformance contract.

Fully removing those test friendships is possible as a separate refactoring: the remaining fixtures would need to resolve public storage interfaces from independently built containers, and the single-server tests need a public composition seam for custom durable stores and proxies. The Immediate.Jobs.Shared friendship with Immediate.Jobs.Benchmarks is unrelated to storage conformance.

## Summary by CodeRabbit

- **New Features**
  - Added a framework-neutral storage-provider conformance catalog covering queues, recurring jobs, graphs, fair queues, and replicas.
  - Added capability detection and support indicators for fair-queue and replica functionality.

- **Bug Fixes**
  - Improved duplicate recurring-job handling and schedule advancement.
  - Strengthened recurring schedule validation, cleanup, and materialization consistency.

- **Documentation**
  - Added comprehensive storage conformance guidance and expanded Testing documentation.

- **Tests**
  - Added conformance coverage across in-memory, relational, single-server, and Redis storage configurations.

### CODE REVIEW

File: src/Immediate.Jobs.EntityFrameworkCore/EntityFrameworkCoreJobStorage.cs [hunk 1]
```diff
@@ -12,10 +12,10 @@ namespace Immediate.Jobs.EntityFrameworkCore;
 /// <typeparam name="TContext">The application context containing the Immediate.Jobs model.</typeparam>
 /// <param name="contextFactory">The factory used to create application database contexts.</param>
 /// <param name="timeProvider">The clock used for storage timestamps, or <see langword="null"/> to use the system clock.</param>
-public sealed class EntityFrameworkCoreJobStorage<TContext>(
+internal sealed class EntityFrameworkCoreJobStorage<TContext>(
 	IDbContextFactory<TContext> contextFactory,
 	TimeProvider? timeProvider = null
-) : IRecurringJobStorage, IJobGraphStorage, IJobStorageReplica
+) : IRecurringJobStorage, IJobGraphStorage, IFairQueueStorage, IJobStorageReplica
```
[the remark this answers is not in the document]
Comment (viceroypenguin) [reply src/Immediate.Jobs.EntityFrameworkCore/EntityFrameworkCoreJobStorage.cs:18]: We're still in preview, am not concerned about breaking consumers between versions. This comment is valid once we hit v1.0.0.

File: src/Immediate.Jobs.Testing/Storage/ReplicaStorageConformance.cs [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+		var second = ConformanceAssert.NotNull(
+			(await replica.AcquireJobsAsync(
+				[job.Id],
+				"stale-worker-two",
+				TimeSpan.FromMinutes(2),
+				cancellationToken
+			).ConfigureAwait(false)).SingleOrDefault(),
+			StaleName,
+			"an expired exact-id lease must be reclaimable"
+		);
```
[the remark this answers is not in the document]
Comment (dukesteen) [reply src/Immediate.Jobs.Testing/Storage/ReplicaStorageConformance.cs:268]: I am keeping the exact-expiry boundary intentionally. The shared contract treats a lease as unexpired only while `LeaseExpiresAt > now` (documented in `docs/fair-queues.md`), so equality is the first expired instant. The in-memory, EF Core, and LINQ to DB implementations all reclaim with `LeaseExpiresAt <= now`, and `docs/job-retention.md` describes the expiry timestamp as the last moment through which ownership was valid. Advancing an extra second would stop this conformance case from testing that boundary and would allow a provider with inconsistent semantics to pass, so no code change is appropriate here.
